### ISSUE TITLE
GM_xmlhttpRequest: `blob.arrayBuffer()` and `blob.text()` return `undefined`

### ISSUE THREAD

**Issue by vabushkevich** [Tampermonkey/tampermonkey#1418]
## Expected Behavior

Inside the `onload` handler when calling the `GM_xmlhttpRequest()` with the `responseType: "blob"` option:

- `e.response.arrayBuffer()` should return a promise that resolves with an `ArrayBuffer`.
- `e.response.text()` should return a promise that resolves with a string.

## Actual Behavior

Both  `e.response.arrayBuffer()` and `e.response.text()`  return `undefined`.

## Specifications

- Chrome: 98.0.4758.102
- TM: 4.14
- OS: Microsoft Windows 10.0.19041.1415

## Script

```js
// ==UserScript==
// @name    GM_xmlhttpRequest issue
// @include *
// @grant   GM_xmlhttpRequest
// @connect *
// ==/UserScript==

(function () {
    'use strict';

    GM_xmlhttpRequest({
        url: "https://www.wikipedia.org",
        method: "GET",
        responseType: "blob",
        onload: (e) => {
            console.log("arrayBuffer():", e.response.arrayBuffer());
            console.log("text():", e.response.text());
        },
        onerror: (err) => {
            console.log("Error:", err);
        },
    });
})();
```

**Comment by derjanb** [comment]
Investigating...

Workaround for now: 

```JS
// ==UserScript==
// @name    GM_xmlhttpRequest issue
// @include *
// @grant   GM_xmlhttpRequest
// @connect *
// ==/UserScript==

(function () {
    'use strict';

    GM_xmlhttpRequest({
        url: "https://www.wikipedia.org",
        method: "GET",
        responseType: "blob",
        onload: (e) => {
            console.log("arrayBuffer():", e.response.arrayBuffer());
            console.log("text():", e.response.text());

            var fileReader = new FileReader();
            fileReader.onload = () => {
                console.log("FileReader.readAsArrayBuffer():", fileReader.result)
            };
            fileReader.readAsArrayBuffer(e.response);

        },
        onerror: (err) => {
            console.log("Error:", err);
        },
    });
})();
```

**Comment by yingziwu** [comment]
Tampermonkey 4.14 also break Blob prototype.

## Script

```javascript
// ==UserScript==
// @name         Blob test
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  Blob test
// @author       You
// @match        http://example.com/
// @grant        GM_xmlhttpRequest
// ==/UserScript==

(async () => {
  const blob = new Blob(['console.log("Hello world.")'], {
    type: 'text/javascript'
  });
  console.log(await blob.arrayBuffer());
  console.log(await blob.text());
})()
```

## Expected Behavior

Console output as follow:

![Expected](https://user-images.githubusercontent.com/22741471/154830806-4dcaa9f6-4f18-423d-8c91-e0a73ae40629.png)

## Actual Behavior

Both return `undefined`.

![Actual](https://user-images.githubusercontent.com/22741471/154830865-15348ee3-b07a-4e97-af86-5e1ca5b12637.png)

**Comment by derjanb** [comment]
Should be fixed at 4.15.6153 (in review|[crx](https://tampermonkey.net/crx/tampermonkey_beta_4_15_6153.crx))
